gptel.el

```diff
@@ -2379,53 +2471,83 @@ be used to rerun or continue the request at a later time."
                  gptel-backend (and gptel--num-messages-to-send
                                     (* 2 gptel--num-messages-to-send))))))
             ((consp prompt) (gptel--parse-list gptel-backend prompt)))))
-         (info (list :data (gptel--request-data gptel-backend full-prompt)
+         (info (list :data (list :args t
```

Comment:
It's not clear to me yet how to avoid this step of storing and rehydrating this state in `:data`.  I'll continue to think about this.  But at minimum I think capturing many of these keys is unnecessary since you pass `info` (actually the whole `fsm`) to the augmentors, and they are available as `:keys` inside `info` in the current implementation on master.

- `:context`: Accessible inside `info` as `:context` anyway.
- `:gptel-include-reasoning`: Available as `:include-reasoning` in `:info`.
- `:callback`: Do we expect the augmentors to change the callback?
- `:in-place`: Available as `:in-place` in `:info`.